taichi/backends/vulkan/aot_module_builder_impl.cpp

```diff
@@ -96,11 +99,33 @@ void AotModuleBuilderImpl::load(const std::string &output_dir) {
       std::vector<uint32_t> res = read_spv_file(output_dir, k.tasks_attribs[j]);
       spirv_sources_codes.push_back(res);
     }
+    ti_aot_data_.spirv_codes.push_back(spirv_sources_codes);
+  }
+}
 
-    VkRuntime::RegisterParams params;
-    params.kernel_attribs = ti_aot_data_.kernels[i];
-    params.task_spirv_source_codes = spirv_sources_codes;
+bool AotModuleBuilderImpl::get_kernel(const std::string &name,
+                                      VkRuntime::RegisterParams &kernel) {
+  for (int i = 0; i < ti_aot_data_.kernels.size(); ++i) {
+    // Kernel names encode more than the name of the function, but for AOT, only
```

Comment:
IIUC this is  more like an offloaded task than a kernel (a kernel (which is ti.kernel in python) consists of many offloaded tasks ;)